File: Go/S1002/S1002.go
package main

import (
	"fmt"
	"math"
)

type person struct {
	x              float64
	y              float64
	targetDistance float64
}

func newPerson() *person {
	p := person{}
	p.x = 0
	p.y = 0
	p.targetDistance = 0
	return &p
}

func main() {
	p1, p2 := newPerson(), newPerson()
	var idx int
	fmt.Scan(&idx)
	for i := 0; i < idx; i++ {
		fmt.Scan(&p1.x, &p1.y, &p1.targetDistance, &p2.x, &p2.y, &p2.targetDistance)
		distance := math.Sqrt(math.Pow(p2.x-p1.x, 2) + math.Pow(p2.y-p1.y, 2))
		sum := p2.targetDistance + p1.targetDistance
		sub := math.Abs(p2.targetDistance - p1.targetDistance)

		if distance == 0 {
			if p1.targetDistance == p2.targetDistance {
				fmt.Println("-1")
			} else {
				fmt.Println("0")
			}
		} else if distance == sum || distance == sub {
			fmt.Println("1")
		} else if distance < sum && distance > sub {
			fmt.Println("2")
		} else if distance > sum || distance < sub {
			fmt.Println("0")
		}
	}
}

File: C/S1002.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

struct Person
{
    int x;
    int y;
    int target_distance;
};

int main(void)
{
    struct Person p1, p2;
    int loopControl = 0;
    scanf("%d", &loopControl);
    while (loopControl > 0)
    {
        scanf("%d %d %d %d %d %d", &p1.x, &p1.y, &p1.target_distance, &p2.x, &p2.y, &p2.target_distance);

        double distance = sqrt(pow(p2.x - p1.x, 2) + pow(p2.y - p1.y, 2));
        int r_sum = p2.target_distance + p1.target_distance;
        int r_sub = abs(p2.target_distance - p1.target_distance);

        if (distance == 0)
        {
            if (p1.target_distance == p2.target_distance)
            {
                printf("-1\n");
            }
            else
            {
                printf("0\n");
            }
        }
        else if (distance == r_sum || distance == r_sub)
        {
            printf("1\n");
        }
        else if (distance < r_sum && distance > r_sub)
        {
            printf("2\n");
        }
        else if (distance > r_sum || distance < r_sub)
        {
            printf("0\n");
        }
        loopControl--;
    }
}